### PR TITLE
feat(claude): add /copilot-pairing and 3-pass review rigor

### PR DESCRIPTION
**Pull request by inkatze**
## Summary

- **New `/copilot-pairing` slash command** (`roles/osx/files/claude/commands/copilot-pairing.md`): autonomous loop of the Copilot review process. Per iteration: validate threads, implement fixes test-first when applicable, run the project's check suite, push, re-request Copilot review, poll up to 10 minutes for the response, repeat. Caps at 10 iterations. Hard stop conditions baked in for ambiguity, scope creep, test failure, security-sensitive changes, loop detection, low-confidence findings, no response, and conflicting signals.
- **Canonical `Validation Rigor` doctrine in `CLAUDE.md`**: two new subsections under `Code & PR Reviews`.
  - *Issue identification*: 3 passes minimum, different angles each. Pass 1 = direct reproduction (failing test, repro script, trace). Pass 2 = orthogonal angle (callers, conventions, sibling implementations, existing coverage). Pass 3 = outside-in (`git log`/`blame`, repo-wide search, and for text/research-based claims: official docs, library source/tests, deepwiki MCP, GitHub issues, RFCs, web search). Drop or downgrade items where passes do not converge.
  - *Solutions*: 2 angles minimum, 3 when relevant. Targeted failing test, wider check (suite + lint + types), then edge cases / integration / manual. Substitute review angles for non-testable changes.
- **All five review skills updated** to reference and apply the doctrine: `self-review`, `copilot-review`, `copilot-pairing`, `peer-review`, `code-review`.
- **`copilot-review` overhaul**: collapsed the old first-pass/second-pass into the canonical 3-pass step, presents findings as a Markdown table (`# | Thread ID | File:Line | Copilot's concern | What we found | Reproduced? | Classification | Confidence | Our proposed fix`), enforces "do not trust Copilot's recommendation, design our own fix" and applies the same rigor to the proposed fix. Replies cite which passes were checked.
- **Workflows section in `CLAUDE.md`** now lists four workflows (was three) including `/copilot-pairing`.

## Implementation notes

- Copilot bot identity confirmed live on `SymmetrySoftware/stl-poc#13`: GraphQL `author { __typename: "Bot", login: "copilot-pull-request-reviewer" }`. Reviews come in as `state: COMMENTED` (not `CHANGES_REQUESTED`). Pairing's done-detection is "Copilot review submitted after our push that creates zero new unresolved threads".
- Pairing's re-request path is mode-aware (set in pre-flight from `__typename`):
  - `app` mode (bot is `__typename: Bot`, e.g. Copilot installed as a GitHub App): the REST endpoint `POST /repos/{owner}/{repo}/pulls/{number}/requested_reviewers` returns 422 "not a collaborator" for App-typed reviewers, so we call the GitHub MCP `request_copilot_review` tool instead (it wraps an internal endpoint that accepts Bot reviewers) and verify by re-querying `reviewRequests.nodes` for the bot login.
  - `collaborator` mode: REST `POST /requested_reviewers` with `reviewers[]=copilot-pull-request-reviewer`, with a DELETE-then-POST retry for the 422 "already requested" case.
  - Either way, step (g)'s 10-minute GraphQL poll for a new Copilot review (filtered by author login plus a baseline-id/timestamp combined predicate) is the authoritative confirmation. Auto-review-on-push has been observed to silently no-op in app mode, so the explicit re-request and the poll are both load-bearing.
- No Ansible task changes needed: `roles/osx/files/claude/commands/` is symlinked wholesale, so the new file is picked up on the next session. Memory file (`MEMORY.md`) updated locally to record the bot login.

## Test plan

- [ ] `readlink ~/.claude/commands/copilot-pairing.md` resolves into `roles/osx/files/claude/commands/copilot-pairing.md`
- [ ] Fresh Claude Code session lists `/copilot-pairing` in available skills
- [ ] `/copilot-review` on a PR with open Copilot threads narrates the 3 passes and produces the table
- [ ] `/copilot-pairing` on a small PR drives at least one full iteration (validate, test-first fix, push, re-request, poll, resume) and a stop condition (ambiguity or iteration cap on contrived input) hands control back cleanly
- [ ] `/self-review`, `/peer-review`, `/code-review` each cite which validation passes converged for at least one finding when run on a real diff
- [ ] CLAUDE.md `Validation Rigor` headings render in the materialized `~/.claude/CLAUDE.md` after the next Ansible run

### CODE REVIEW

File: roles/osx/files/claude/commands/code-review.md [hunk 1]
```diff
@@ -41,16 +41,24 @@ If the diff is large, review file-by-file using `git diff <base>...HEAD -- <path
...
+
+- **Read first.** The full source file to understand context, plus callers and related modules.
+- **Pass 1: direct reproduction.** When the issue concerns runtime behavior, reproduce it. Failing test, repro script, trace through the code with concrete inputs, or construct an input that triggers the bug. Inability to reproduce is a strong signal of a false positive.
+- **Pass 2: orthogonal angle.** A different lens: callers and what they assume, related code paths and side effects, project conventions, sibling implementations, existing test coverage.
+- **Pass 3: outside-in angle.** Sources outside the diff: `git log` / `git blame` for the why-it-is-the-way-it-is, repo-wide search for similar patterns, and for text/research-based claims (API correctness, spec compliance, deprecated patterns, security claims, library behavior) consult official docs, the library's own source/tests, deepwiki MCP, GitHub issues, RFCs, web search. Note what was checked.
+
+Drop or downgrade items where the three passes do not converge. We are leaving comments on someone else's PR — false positives have a higher cost here than in self-review.
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/code-review.md]: addressed in 0fb9ed6 (em-dashes replaced with colons / commas / parens / periods to match the CLAUDE.md writing-style rule)

File: roles/osx/files/CLAUDE.md [hunk 2]
```diff
@@ -96,12 +96,31 @@ When reviewing code, features, or addressing PR feedback:
...
+There are four distinct review workflows, each with a corresponding slash command:
 
 1. **Self-review** (`/self-review`): Comprehensive code review of the feature branch against main. Review, validate for false positives, iterate until clean, then push and create a draft PR.
-2. **Copilot review** (`/copilot-review`): Address unresolved GitHub Copilot review threads on the current PR. Fetch threads via GraphQL, validate each one (emphasis on catching false positives), address valid items, then comment and resolve threads via GraphQL.
-3. **Peer review** (`/peer-review`): Address unresolved peer review threads on the current PR. Same validation process as Copilot review, but responses must sound natural, human, and match the user's communication style.
+2. **Copilot review** (`/copilot-review`): Address unresolved GitHub Copilot review threads on the current PR. Fetch threads via GraphQL, reproduce each issue when relevant, design our own fix (do not trust Copilot's recommendation), do a second-pass validation, present findings as a table, then implement test-first when applicable, comment, and resolve threads via GraphQL.
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/CLAUDE.md]: Fixed in 173fcf7. The `/copilot-review` workflow summary in CLAUDE.md now says "validate via the three-pass rigor" instead of "do a second-pass validation", aligning the description with the canonical Validation Rigor (Issue Identification) doctrine added in the same diff.

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+   gh api graphql -f query='
+     query($owner: String!, $repo: String!, $number: Int!) {
+       repository(owner: $owner, name: $repo) {
+         pullRequest(number: $number) {
+           reviews(first: 5) { nodes { author { __typename login } } }
+         }
+       }
+     }
+   ' -f owner='OWNER' -f repo='REPO' -F number=NUMBER
+   ```
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md]: Fixed in 173fcf7. Switched both `/copilot-pairing` pre-flight step 3 and the equivalent block in `/copilot-review` step 3 from `reviews(first: 5)` to `reviews(last: 5)` so the bot-login verification surfaces the most-recent reviews rather than the oldest. Same fix applied in two places because both files are in this PR diff and would otherwise drift.

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+   - Iteration counter = 0.
+
+## Iteration loop
+
+For each iteration (cap = **10**):
+
+### a. Fetch Copilot's open threads
+
+Use the same GraphQL query as `/copilot-review` step 3. Filter to threads where `isResolved: false` AND first comment author login == Copilot bot login. If a thread's first comment was created before the last commit SHA we pushed, skip it (it predates the latest review and may be stale).
+
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md]: addressed in 52a32e3 (the SHA-based stale-skip filter was dropped; step (a) now relies on `isResolved: false` as the canonical signal and explicitly says `Do not add a "skip if older than last push" filter`)

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+For **false positives**, draft the dismissal reply (citing the three passes) but do NOT post yet. We post all replies in step (e) after the build is green.
+
+### d. Run the full local check suite
+
+Run whatever the project ships for local verification: tests, linters, type checkers, formatters. Common entry points: `npm test`, `pytest`, `go test ./...`, `cargo test`, `bundle exec rspec`, `mise run test`, `lefthook run pre-commit`. If the project has a single canonical command, prefer it. If anything fails — even a pre-existing failure unrelated to our changes — trigger the **Test failure** stop condition.
+
+### e. Reply, resolve, commit, push
+
+- Reply to and resolve every thread using `/copilot-review` step 9 mutations (multi-line GraphQL, body via temp file).
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md]: addressed in 0fb9ed6 (cross-reference corrected: `/copilot-pairing` step (e) now points at `/copilot-review` step 8, which is the actual reply/resolve step)

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+gh api -X POST "repos/OWNER/REPO/pulls/NUMBER/requested_reviewers" \
+  -f 'reviewers[]=copilot-pull-request-reviewer'
+```
+
+If the request returns 422 because Copilot is already a requested reviewer, first remove it then re-add:
+
+```bash
+gh api -X DELETE "repos/OWNER/REPO/pulls/NUMBER/requested_reviewers" \
+  -f 'reviewers[]=copilot-pull-request-reviewer'
+```
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md]: addressed in d951dca (and refined in 47b1e3f). Both the `collaborator`-mode re-request example and the step (g) poll script now say "Substitute the verified bot login from pre-flight step 3 if it differs from the default" wherever the example shows the default `copilot-pull-request-reviewer` login.

File: roles/osx/files/claude/commands/copilot-review.md [hunk 7]
```diff
@@ -45,25 +45,51 @@ gh api graphql -f query='
...
+**Do not take Copilot's recommendation as correct.** Even when the underlying concern is real, design the best solution from first principles. Copilot's suggested fix may be insufficient (treating a symptom not the cause), wrong (introduces a new bug), unidiomatic for the codebase, or out of scope. Apply the same three-pass rigor to the proposed fix: does it actually resolve the issue, does it survive an orthogonal angle, does it match what docs/conventions/external references would recommend.
 
-### 6. Address items
+Classify each thread as **valid** (needs a fix), **false positive** (no real problem), or **low-confidence** (passes did not converge — never guess). Look for issues Copilot did not flag that are adjacent to what it did. Surface those as extra rows tagged "adjacent finding".
+
+### 5. Present the validated table
+
+Output one Markdown table. Default columns:
+
+| # | Thread ID | File:Line | Copilot's concern | What we found | Reproduced? | Classification | Confidence | Our proposed fix |
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-review.md:91]: Fixed in 173fcf7. Added the `|---|---|---|---|---|---|---|---|---|` separator row beneath the 9-column header so the example renders as a GitHub-flavored Markdown table.

File: roles/osx/files/claude/commands/peer-review.md [hunk 8]
```diff
@@ -47,20 +47,22 @@ gh api graphql -f query='
 ' -f owner='OWNER' -f repo='REPO' -F number=NUMBER
 ```
 
-### 4. Validate each unresolved thread
+### 4. Validate each unresolved thread — three passes minimum (different angle each)
 
-Filter to only unresolved threads (`isResolved: false`). For each one:
+Filter to only unresolved threads (`isResolved: false`). For each one, apply the canonical rigor in CLAUDE.md `Validation Rigor (Issue Identification)`:
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/peer-review.md]: addressed in 47b1e3f (heading "### 4. Validate each unresolved thread — three passes minimum" became "### 4. Validate each unresolved thread: three passes minimum"; em-dash to colon to match the CLAUDE.md writing-style rule, in line with the same dash-removal pass applied to other commands)

File: roles/osx/files/claude/commands/copilot-review.md [hunk 9]
```diff
@@ -45,37 +45,97 @@ gh api graphql -f query='
...
-- Read the comment body and identify the referenced file path and line(s)
-- Read the actual source code at that location
-- **Validate thoroughly**: Determine if the concern is legitimate by checking the real code. Copilot often produces false positives.
-- Classify as: **valid** (needs a fix), **false positive**, or **low-confidence**
+Concretely, pipe the reviewThreads response into jq (substitute the verified bot login if it isn't the default):
+
+```bash
+| jq --arg bot 'copilot-pull-request-reviewer' '
+    .data.repository.pullRequest.reviewThreads.nodes
+    | map(select(.isResolved == false and .comments.nodes[0].author.login == $bot))'
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-review.md]: Fixed in b18d9fb. Dropped the leading pipe and reworded the prose to "Pipe the previous query's output through this jq filter (...)". The snippet is now a syntactically valid `jq` invocation when copy-pasted (jq accepts the previous output via stdin, e.g. `gh api graphql ... | jq ...`), and the prose still keeps the GraphQL query from being duplicated.

File: roles/osx/files/claude/commands/peer-review.md [hunk 10]
```diff
@@ -47,20 +47,31 @@ gh api graphql -f query='
...
 
-- Read the full comment thread (all comments, not just the first)
-- Read the actual source code at the referenced location
-- **Validate carefully**: Determine if the concern is legitimate, a false positive, or a matter of preference
-- Consider the reviewer's perspective and intent
+Concretely, pipe the reviewThreads response into jq:
+
+```bash
+| jq '.data.repository.pullRequest.reviewThreads.nodes
+    | map(select(.isResolved == false and .comments.nodes[0].author.__typename != "Bot"))'
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/peer-review.md]: Fixed in b18d9fb. Same fix applied here as in `/copilot-review`: dropped the leading pipe, reworded the prose to "Pipe the previous query's output through this jq filter:". The snippet is now a syntactically valid `jq` invocation; in practice it is run as `<previous gh api graphql command> | <this jq filter>`.

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 11]
```diff
@@ -0,0 +1,293 @@
...
+      repository(owner: $owner, name: $repo) {
+        pullRequest(number: $number) {
+          reviews(last: 5) { nodes { author { login } state submittedAt } }
+        }
+      }
+    }
+  ' -f owner='OWNER' -f repo='REPO' -F number=NUMBER \
+    | jq -r --arg bot 'copilot-pull-request-reviewer' --argjson since "$push_epoch" '
+        .data.repository.pullRequest.reviews.nodes
+        | map(select(.author.login == $bot and (.submittedAt | sub("\\.[0-9]+Z$"; "Z") | fromdateiso8601) > $since))
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md]: Fixed in 85d56d4. Step (e) now captures a baseline most-recent-Copilot-review id alongside `push_epoch`, and the step (g) poll filter combines two predicates: `id != baseline_review_id` (excludes the prior review) and `submittedAt >= push_epoch` (excludes pre-existing older Copilot reviews). The combined filter unambiguously matches a same-second submission while still rejecting older reviews. We considered the simpler "id != baseline" alone but rejected it: when prior Copilot reviews exist, an older review (still within the last-5 window) would also have an id different from the baseline and would match. We considered nanosecond-precision push_epoch but rejected it because macOS `date` does not support `%N`. The combined predicate is portable and removes the boundary bug.

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 12]
```diff
@@ -0,0 +1,293 @@
...
+
+This call is **best-effort**, not load-bearing. The actual signal the loop relies on is step (g)'s GraphQL poll for a new review.
+
+Inspect the response and branch:
+
+| Outcome | Body contains | Action |
+|---|---|---|
+| 2xx | n/a | Proceed to step (g). |
+| 422 | `already requested` (or similar "duplicate reviewer") | DELETE the reviewer, re-POST. Then proceed to step (g). |
+| 422 | `not a collaborator` | Pre-flight mode detection was wrong (the bot is no longer a Bot-typed reviewer). Log the mismatch and proceed to step (g). On the next run, re-check pre-flight step 3. |
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md]: Fixed in 85d56d4. The "not a collaborator" 422 row no longer asserts "pre-flight mode detection was wrong" as the only cause. It now lists the possible causes (App-typed bot that pre-flight misclassified, lost collaborator status mid-run, login otherwise ineligible) and routes the same proceed-to-step-(g) action. The poll is the authoritative signal regardless of the cause, and pre-flight is still re-checked on the next run.

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 13]
```diff
@@ -0,0 +1,322 @@
...
+        .data.repository.pullRequest.reviews.nodes
+        | map(select(
+            .author.login == $bot
+            and .id != $baseline
+            and (.submittedAt | sub("\\.[0-9]+Z$"; "Z") | fromdateiso8601) >= $since
+          ))
+        | last // empty')
+  if [ -n "$latest" ]; then echo "NEW_REVIEW $latest"; exit 0; fi
+  sleep 30
+done
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md]: Fixed in fb49d64. Reworded the step (g) preface to make clear the harness constraint applies at the tool-invocation boundary: long leading sleeps are blocked at the start of a Bash tool call, and shorter sleeps chained across multiple tool calls are also blocked. A single backgrounded Bash invocation that sleeps internally is fine because the harness does not introspect a long-running backgrounded process. The script in the example is consistent with that rule.

File: roles/osx/files/claude/commands/copilot-review.md [hunk 14]
```diff
@@ -45,37 +45,97 @@ gh api graphql -f query='
...
-- Read the comment body and identify the referenced file path and line(s)
-- Read the actual source code at that location
-- **Validate thoroughly**: Determine if the concern is legitimate by checking the real code. Copilot often produces false positives.
-- Classify as: **valid** (needs a fix), **false positive**, or **low-confidence**
+Pipe the previous query's output through this jq filter (substitute the verified bot login if it isn't the default):
+
+```bash
+jq --arg bot 'copilot-pull-request-reviewer' '
+    .data.repository.pullRequest.reviewThreads.nodes
+    | map(select(.isResolved == false and .comments.nodes[0].author.login == $bot))'
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-review.md]: Fixed in 123ec12. The prose now reads "Apply this jq filter to the `reviewThreads` query output (the first GraphQL block in this step, not the `reviews(last: 5)` bot-login verification block)". `/peer-review` keeps the original "Pipe the previous query's output" wording because it has no intervening query so there is no ambiguity there.

File: roles/osx/files/claude/commands/peer-review.md [hunk 15]
```diff
@@ -47,20 +47,31 @@ gh api graphql -f query='
 ' -f owner='OWNER' -f repo='REPO' -F number=NUMBER
 ```
 
-### 4. Validate each unresolved thread
+### 4. Validate each unresolved thread: three passes minimum (different angle each)
 
-Filter to only unresolved threads (`isResolved: false`). For each one:
+Filter to threads where `isResolved: false` AND the first comment author is NOT a `Bot` (`__typename != "Bot"`). Bot threads (Copilot and friends) belong to `/copilot-review` or `/copilot-pairing`; processing them here would generate human-tone replies to a bot.
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/peer-review.md:53]: Fixed in 8d7c0a4 with a docs-only "Known gap" note rather than narrowing the filter or adding a new workflow. Reasoning across the three passes: (1) Reading: the existing `__typename != "Bot"` filter in /peer-review and the login-based filter in /copilot-review and /copilot-pairing are deliberate. /copilot-review explicitly says it filters on login "so other bot integrations (CodeQL, Dependabot review, etc.) don't get pulled in". The gap is real: non-Copilot bot threads end up unhandled. (2) Orthogonal: narrowing /peer-review's filter to only the Copilot login would route non-Copilot bot threads into a human-tone workflow, which is exactly what the original filter avoided. Adding a new full workflow for non-Copilot bots (option (b) of the suggestion) is non-trivial and would need its own tone instructions per bot integration. (3) Outside-in: the simplest action that addresses the silent-ignoring concern without changing behaviour is to make the gap explicit and tell the reader how to handle it manually or by copying the command. The new note names the trade-off and points to the two practical paths.

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 16]
```diff
@@ -0,0 +1,322 @@
...
+   - `git add` only the files we actually changed for this iteration (never `git add -A`).
+   - Commit with a message of the form `chore(copilot): iter N, address <short summary>`.
+   - Push: `git push origin <branch>`. **Never** `--force`, `--force-with-lease`, or any rebase flag.
+2. **Capture push timestamp and baseline Copilot-review id** (substitute the PR number from pre-flight step 1):
+   ```bash
+   date +%s > /tmp/copilot-pairing-push-epoch.NUMBER
+   gh api graphql -f query='
+     query($owner: String!, $repo: String!, $number: Int!) {
+       repository(owner: $owner, name: $repo) {
+         pullRequest(number: $number) {
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md:90]: Classified as false positive after the three-pass validation. (1) Reproduction: the race requires Copilot to submit a review within ~milliseconds of receiving the push event. Push acknowledgment plus Copilot's LLM-driven review pipeline takes seconds-to-minutes in observed runs (this very PR consistently shows ~5-7 minute review latency). The race window cannot be triggered in practice. (2) Orthogonal angle: the suggested fix (capture push_epoch and baseline_id *before* pushing) opens a wider race in the opposite direction. Between pre-push capture and push, an unrelated review submitted post-capture from any prior request would have an id != baseline and submittedAt >= push_epoch, producing a false-positive NEW_REVIEW (we exit the poll thinking Copilot reviewed our change when it actually reviewed something earlier). The proposed fix is worse than the current code. (3) Outside-in: in app mode (this PR), the doc explicitly notes auto-review-on-push silently no-ops. Copilot only reviews after the explicit `request_copilot_review` MCP call in step (f), which runs well after the post-push `push_epoch` capture. The race scenario does not apply. Leaving the current order unchanged.

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 17]
```diff
@@ -0,0 +1,322 @@
+Iterate with GitHub Copilot autonomously until it stops requesting changes on the current PR.
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md]: Fixed in cacbf0b. Reworded the opening line to "until it has no unresolved threads on the current PR" and added a clarifying sentence: "Copilot reviews land as `state: COMMENTED` (not `CHANGES_REQUESTED`), so the termination signal is the latest Copilot review after our push leaves zero unresolved Copilot threads, not a state change." That matches step (g)'s success branch ("If zero unresolved, success: exit the loop") and step (f.5)'s zero-remaining short-circuit.

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 18]
```diff
@@ -0,0 +1,322 @@
...
+         pullRequest(number: $number) {
+           reviews(last: 5) { nodes { author { __typename login } } }
+         }
+       }
+     }
+   ' -f owner='OWNER' -f repo='REPO' -F number=NUMBER
+   ```
+   Use the actual `Bot` login if it differs. Then set `repo_mode` from the same `__typename` field:
+   - `__typename == "Bot"` → `repo_mode = "app"`. Copilot is installed as a GitHub App, not a collaborator. The REST endpoint `POST /repos/{owner}/{repo}/pulls/{n}/requested_reviewers` returns 422 ("not a collaborator") for App-typed reviewers and must NOT be used in this mode. Step (f) instead calls the `request_copilot_review` MCP tool (which wraps an internal endpoint that accepts Bot reviewers). Do not assume push alone will trigger a Copilot review: auto-review-on-push has been observed to silently no-op (see step (f) for the verified failure mode), so step (g)'s 10-minute poll is the only authoritative confirmation that Copilot has reviewed.
+   - Otherwise → `repo_mode = "collaborator"`. Use the explicit re-request POST in step (f).
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md:27]: Updated the PR description rather than reverting the command. The MCP-based app-mode path is deliberate: REST `POST /requested_reviewers` returns 422 "not a collaborator" for App-typed bots (verified live on `SymmetrySoftware/stl-poc#13` 2026-05-02), so the MCP `request_copilot_review` tool is required there. The PR description now lists both modes explicitly: app mode uses the MCP tool plus a `reviewRequests.nodes` verification, collaborator mode uses REST `POST /requested_reviewers` with the DELETE-then-POST 422 retry. Either way, step (g)'s 10-minute GraphQL poll is the authoritative confirmation that Copilot reviewed.

File: roles/osx/files/claude/commands/peer-review.md [hunk 19]
```diff
@@ -107,6 +137,42 @@ gh api graphql -f query='
...
+
+Query the PR's pending reviews and the viewer's login in one round-trip, then submit each pending review owned by the viewer:
+
+```bash
+gh api graphql -f query='
+  query($owner: String!, $repo: String!, $number: Int!) {
+    viewer { login }
+    repository(owner: $owner, name: $repo) {
+      pullRequest(number: $number) {
+        reviews(states: PENDING, first: 10) {
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/peer-review.md:150]: Classified as false positive after the three-pass validation. (1) Reproduction: the exact query `reviews(states: PENDING, first: 10) { ... }` has been executed against the live GitHub GraphQL endpoint seven times in the current /copilot-pairing run (the auto-vivified-pending check after every iteration's reply batch) and returned a valid `{"data": ..., "nodes": []}` response every time, no schema error. (2) Orthogonal angle: this same pattern is the documented auto-vivified-pending-review check across `/copilot-review` step 8, `/copilot-pairing` step (e.4), and `/peer-review` step 8. It has been working unbroken. (3) Outside-in: GraphQL spec § 3.6 (Input List Value Coercion) defines that a single non-list value supplied for a list-typed argument is coerced to a single-element list at runtime. GitHub's GraphQL implementation follows this rule, so `states: PENDING` is valid syntax for an argument typed `[PullRequestReviewState!]`. Leaving the wording as-is. If we ever switched to `states: [PENDING]` it would be a cosmetic change (functionally equivalent), not a bug fix.

File: roles/osx/files/claude/commands/copilot-pairing.md [hunk 20]
```diff
@@ -0,0 +1,322 @@
...
+   - **`false positive`**: post the dismissal reply drafted in step (b) (citing the three passes and why the concern does not apply).
+4. **Submit any auto-vivified pending review (mandatory).** `addPullRequestReviewThreadReply` can create a new pending review owned by the viewer when none is in progress; the replies posted in (e.3) then stay invisible (to GitHub UI, to Copilot, to humans) until that review is submitted. See `/copilot-review` step 8's "Submit any auto-vivified pending review" sub-step for the exact GraphQL. Procedure: query the PR's `reviews(states: PENDING)` filtered by `author.login == viewer.login`; for each, call `submitPullRequestReview(id, event: COMMENT)`. Re-query and assert zero pending reviews owned by the viewer remain before proceeding to step (e.5). If a pending review cannot be submitted, stop with the **Pending reply unsubmittable** condition.
+5. **Resolve threads** using `/copilot-review` step 8's `resolveReviewThread` mutation.
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-pairing.md:113]: Classified as false positive after the three-pass validation. (1) Reproduction: the exact query `reviews(states: PENDING, first: 10) { ... }` has been executed against the live GitHub GraphQL endpoint seven times in the current /copilot-pairing run (the auto-vivified-pending check after every iteration's reply batch) and returned a valid `{"data": ..., "nodes": []}` response every time, no schema error. (2) Orthogonal angle: this same pattern is the documented auto-vivified-pending-review check across `/copilot-review` step 8, `/copilot-pairing` step (e.4), and `/peer-review` step 8. It has been working unbroken. (3) Outside-in: GraphQL spec § 3.6 (Input List Value Coercion) defines that a single non-list value supplied for a list-typed argument is coerced to a single-element list at runtime. GitHub's GraphQL implementation follows this rule, so `states: PENDING` is valid syntax for an argument typed `[PullRequestReviewState!]`. Leaving the wording as-is. If we ever switched to `states: [PENDING]` it would be a cosmetic change (functionally equivalent), not a bug fix.

File: roles/osx/files/claude/commands/copilot-review.md [hunk 21]
```diff
@@ -93,6 +153,42 @@ gh api graphql -f query='
...
+
+Query the PR's pending reviews and the viewer's login in one round-trip, then submit each pending review owned by the viewer:
+
+```bash
+gh api graphql -f query='
+  query($owner: String!, $repo: String!, $number: Int!) {
+    viewer { login }
+    repository(owner: $owner, name: $repo) {
+      pullRequest(number: $number) {
+        reviews(states: PENDING, first: 10) {
```
[the remark this answers is not in the document]
Comment (inkatze) [reply roles/osx/files/claude/commands/copilot-review.md:166]: Classified as false positive after the three-pass validation. (1) Reproduction: the exact query `reviews(states: PENDING, first: 10) { ... }` has been executed against the live GitHub GraphQL endpoint seven times in the current /copilot-pairing run (the auto-vivified-pending check after every iteration's reply batch) and returned a valid `{"data": ..., "nodes": []}` response every time, no schema error. (2) Orthogonal angle: this same pattern is the documented auto-vivified-pending-review check across `/copilot-review` step 8, `/copilot-pairing` step (e.4), and `/peer-review` step 8. It has been working unbroken. (3) Outside-in: GraphQL spec § 3.6 (Input List Value Coercion) defines that a single non-list value supplied for a list-typed argument is coerced to a single-element list at runtime. GitHub's GraphQL implementation follows this rule, so `states: PENDING` is valid syntax for an argument typed `[PullRequestReviewState!]`. Leaving the wording as-is. If we ever switched to `states: [PENDING]` it would be a cosmetic change (functionally equivalent), not a bug fix.
